app/javascript/packages/document-capture/components/in-person-full-address-search.tsx

```diff
@@ -0,0 +1,227 @@
+import { TextInput } from '@18f/identity-components';
+import { useState, useRef, useEffect, useCallback } from 'react';
+import { t } from '@18f/identity-i18n';
+import { request } from '@18f/identity-request';
+import ValidatedField from '@18f/identity-validated-field/validated-field';
+import SpinnerButton, { SpinnerButtonRefHandle } from '@18f/identity-spinner-button/spinner-button';
+import type { RegisterFieldCallback } from '@18f/identity-form-steps';
+import useSWR from 'swr/immutable';
+import { useDidUpdateEffect } from '@18f/identity-react-hooks';
+import {
+  FormattedLocation,
+  transformKeys,
+  snakeCase,
+  LocationQuery,
+  LOCATIONS_URL,
+  PostOffice,
+} from '@18f/identity-address-search';
+
+const formatLocations = (postOffices: PostOffice[]): FormattedLocation[] =>
+  postOffices.map((po: PostOffice, index) => ({
+    formattedCityStateZip: `${po.city}, ${po.state}, ${po.zip_code_5}-${po.zip_code_4}`,
+    id: index,
+    distance: po.distance,
+    name: po.name,
+    saturdayHours: po.saturday_hours,
+    streetAddress: po.address,
+    sundayHours: po.sunday_hours,
+    weekdayHours: po.weekday_hours,
+    isPilot: !!po.is_pilot,
+  }));
+
+const requestUspsLocations = async (address: LocationQuery): Promise<FormattedLocation[]> => {
+  const response = await request<PostOffice[]>(LOCATIONS_URL, {
+    method: 'post',
+    json: { address: transformKeys(address, snakeCase) },
+  });
+
+  return formatLocations(response);
+};
+
+function useUspsLocations() {
+  const [locationQuery, setLocationQuery] = useState<LocationQuery | null>(null);
+  const validatedAddressFieldRef = useRef<HTMLFormElement>(null);
+  const validatedCityFieldRef = useRef<HTMLFormElement>(null);
+  const validatedStateFieldRef = useRef<HTMLFormElement>(null);
+  const validatedZipCodeFieldRef = useRef<HTMLFormElement>(null);
+
+  const handleLocationSearch = useCallback(
+    (event, addressInput, cityInput, stateInput, zipCodeInput) => {
+      event.preventDefault();
+      validatedAddressFieldRef.current?.setCustomValidity('');
+      validatedAddressFieldRef.current?.reportValidity();
+      validatedCityFieldRef.current?.setCustomValidity('');
+      validatedCityFieldRef.current?.reportValidity();
+      validatedStateFieldRef.current?.setCustomValidity('');
+      validatedStateFieldRef.current?.reportValidity();
+      validatedZipCodeFieldRef.current?.setCustomValidity('');
+      validatedZipCodeFieldRef.current?.reportValidity();
```

Comment:
Is there some custom validation errors we're trying to reset with these calls? I don't see where we're doing that custom validation.

Reply:
I created [LG-10435](https://cm-jira.usa.gov/browse/LG-10435) to track this suggestion.